src/Microsoft.Data.SqlClient/extensions/Extensions/src/ISqlAuthenticationParameters.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.Data.SqlClient;
+
+/// <include file='../doc/ISqlAuthenticationParameters.xml' path='docs/members[@name="ISqlAuthenticationParameters"]/ISqlAuthenticationParameters/*'/>
+public interface ISqlAuthenticationParameters
```

Comment:
@paulmedynski You may want to reconsider the usage of an interface for a public API. 
Interfaces cannot be evolved with new members. 
If a new method/property is added to interface, then older implementations become incompatible. With base class and virtual methods, the evolution of public API is possible. 

I thought I will leave this thought here, even though I know this PR is WIP.